### PR TITLE
Merge in contract updates

### PR DESCRIPTION
**Pull request by voltrevo**
## What is this PR doing?

Merges in changes from `contract-updates`.

I needed to add a few changes to make this work:

1. Resolve a merge conflict with `getOperationResults`
2. Fix linting error due to deno deprecating `Deno.run`
3. Update deployments on arbitrum-goerli and optimism-goerli
4. Update `bls-wallet-clients` in aggregator and extension
5. Remove assertions about the `to` address during test-integration (breaking because the aggregator now uses the entry point rather than `VerificationGateway` at the top level, and these tests didn't really need to assert the to address anyway)

You can see the diff for (2)-(5) here:
https://github.com/web3well/bls-wallet/compare/0047f14...merge-in-contract-updates

For (1), this is the merge conflict I resolved:

<details>
<summary>Click to expand</summary>

```merge
/**
 * Gets the results of operations (and actions) run through VerificationGateway.processBundle.
 * Decodes unsuccessful operations into an error message and the index of the action that failed.
 *
 * @param transactionReceipt Transaction receipt from a VerificationGateway.processBundle transaction
 * @returns An array of decoded operation results
 */
export const getOperationResults = (
  txnReceipt: ContractReceipt,
): OperationResult[] => {
<<<<<<< HEAD
  if (!txnReceipt.events?.length) {
    throw new Error(
      `no events found in transaction ${txnReceipt.transactionHash}, are you sure it is a ContractReceipt from a VerificationGateway.processBundle transaction?`,
    );
  }

  const walletOpProcessedEvents = txnReceipt.events?.filter(
    (e) => e.event === "WalletOperationProcessed",
  );
  if (!walletOpProcessedEvents?.length) {
||||||| 638e1a1f
  const walletOpProcessedEvents = txnReceipt.events?.filter(
    (e) => e.event === "WalletOperationProcessed",
  );
  if (!walletOpProcessedEvents?.length) {
=======
  let walletOpProcessedEvents: Result[];

  if (txnReceipt.events !== undefined) {
    walletOpProcessedEvents = txnReceipt.events
      .filter((e) => e.event === "WalletOperationProcessed")
      .map(({ args }) => {
        if (!args) {
          throw new Error("WalletOperationProcessed event missing args");
        }

        return args;
      });
  } else if (txnReceipt.logs !== undefined) {
    const vgInterface = VerificationGatewayFactory.createInterface();

    const WalletOperationProcessed = vgInterface.getEvent(
      "WalletOperationProcessed",
    );

    walletOpProcessedEvents = txnReceipt.logs
      .filter(
        (log) =>
          log.topics[0] === vgInterface.getEventTopic(WalletOperationProcessed),
      )
      .map((log) =>
        vgInterface.decodeEventLog(
          WalletOperationProcessed,
          log.data,
          log.topics,
        ),
      );
  } else {
    walletOpProcessedEvents = [];
  }

  if (walletOpProcessedEvents.length === 0) {
>>>>>>> origin/contract-updates
    throw new Error(
      `no WalletOperationProcessed events found in transaction ${txnReceipt.transactionHash}`,
    );
  }

  return walletOpProcessedEvents.reduce<OperationResult[]>(
    (opResults, { wallet, nonce, actions: rawActions, success, results }) => {
      const actions = rawActions.map(
        ({
          ethValue,
          contractAddress,
          encodedFunction,
        }: {
          ethValue: BigNumber;
          contractAddress: string;
          encodedFunction: string;
        }) => ({
          ethValue,
          contractAddress,
          encodedFunction,
        }),
      );
      const error = getError(success, results);

      const ret = [
        ...opResults,
        {
          walletAddress: wallet,
          nonce,
          actions,
          success,
          results,
          error,
        },
      ];

      return ret;
    },
    [],
  );
};
```

</details>

I resolved this by clicking 'Accept Incoming Change' and also deleting the test `fails if no events are in transaction`.

The reason for this conflict is that I ran into a similar issue to this one: https://github.com/web3well/bls-wallet/pull/583 (during compression work, which has already been reviewed+merged into contract-updates). The problem in my case was that using `sendTransaction` directly (instead of `.processBundle`) doesn't generate the `.events` property on the `ContractReceipt`. I implemented a more complete fix to this problem by processing the `.logs` property to get the same information, which is still there when using this lower-level API.

For this reason, it isn't appropriate to throw an error when `.events` is missing, which is what the test I deleted was checking for. There is still a similar test called `fails if no WalletOperationProcessed events are in transaction` which didn't need any modification.

## How can these changes be manually tested?

Run all the tests (covered by CI), also run an aggregator and the extension and make sure you can do an ETH transfer.

(EDIT: We found out these instructions are **wrong**. Oops.)
Additionally, you could check that I resolved the merge conflict as stated above like this:
1. `git fetch origin`
2. `git checkout 'origin/main@{2023-05-03T06:32:14.306Z}'` (checks out the commit from `main` at that timestamp)
3. `git merge 'origin/contract-updates@{2023-05-03T06:32:14.306Z}'`
4. Resolve the conflict in `OperationResults.ts` by clicking 'Accept Incoming Change' in vs code (or by manually accepting the `origin/contract-updates@{2023-05-03T06:32:14.306Z}` version, deleting the other section(s))
5. Delete the test called `fails if no events are in transaction`
6. `git add .`
7. `git commit`
8. Now run `git diff 0047f14` and verify that the output is empty

This confirms that `0047f14` was formed from merging `contract-updates` into `main` (and all those changes have already been reviewed) plus the merge resolution that I specified.

It is therefore appropriate to review this PR as the validity of that merge resolution plus the diff generated by `git diff 0047f14...origin/merge-in-contract-updates` (also viewable on github using [this link](https://github.com/web3well/bls-wallet/compare/0047f14...merge-in-contract-updates) provided previously).

## Does this PR resolve or contribute to any issues?

Nope.

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

File: contracts/clients/src/OperationResults.ts [hunk 1]
```diff
@@ -117,28 +119,49 @@ const getError = (
 export const getOperationResults = (
   txnReceipt: ContractReceipt,
 ): OperationResult[] => {
-  if (!txnReceipt.events?.length) {
-    throw new Error(
-      `no events found in transaction ${txnReceipt.transactionHash}, are you sure it is a ContractReceipt from a VerificationGateway.processBundle transaction?`,
+  let walletOpProcessedEvents: Result[];
+
+  if (txnReceipt.events !== undefined) {
```
Comment (JohnGuilding) [inline contracts/clients/src/OperationResults.ts:124]: Not a blocking comment, but is it worth adding a test for this new logic? e.g. `decodes WalletOperationProcessed logs`. At the moment, we only have a positive test case for a `ContractReceipt` with events, not with logs.

**Review by JohnGuilding — APPROVED** [review]
Diff looks good to me. Was able to send ETH via Quill on the geth local network too.

I wasn't able to follow the steps for verifying the merge conflicts were resolved as mentioned, but had a close look at the `OperationResult.ts` diff and happy with it. After running `git merge 'origin/contract-updates@{2023-05-03T06:32:14.306Z}'`, I got no merge conflicts in `OperationResults.ts`. Not sure if I was doing anything wrong there, just followed the commands as described. Started from main and had latest for both `main` and `merge-in-contract-updates`.

As a side note, the gas savings results from the compression work is really neat.

**Comment by voltrevo** [comment]
> I wasn't able to follow the steps for verifying the merge conflicts were resolved as mentioned, but had a close look at the `OperationResult.ts` diff and happy with it. After running `git merge 'origin/contract-updates@{2023-05-03T06:32:14.306Z}'`, I got no merge conflicts in `OperationResults.ts`. Not sure if I was doing anything wrong there, just followed the commands as described. Started from main and had latest for both `main` and `merge-in-contract-updates`.

Not your fault, I made an oops. Turns out that the branch@{time} syntax means what your local clone's view of that branch was at that time, rather than the commit for that branch according to the latest history.

I've learnt now what (I think 😅) is the correct solution to what I was trying to do. Here's the updated instructions if you're curious:

1. `git fetch origin`
2. `git checkout $(git rev-list --first-parent -1 --before '2023-05-03T06:32:14.306Z' origin/main)` (checks out the commit from `main` at that timestamp)
3. `git merge $(git rev-list --first-parent -1 --before '2023-05-03T06:32:14.306Z' origin/contract-updates)`
4. Resolve the conflict in `OperationResults.ts` by clicking 'Accept Incoming Change' in vs code (or by manually accepting the `0d52ddb20f2a395f3fff13130459a8561533024b` version, deleting the other section(s))
5. Delete the test called `fails if no events are in transaction`
6. `git add .`
7. `git commit`
8. Now run `git diff 0047f14` and verify that the output is empty

> As a side note, the gas savings results from the compression work is really neat.

Thanks! 💪

File: contracts/clients/src/OperationResults.ts (hunk 1 above)
Comment (voltrevo) [reply contracts/clients/src/OperationResults.ts:124]: Yeah good point. I found another little issue by doing this and fixed it too: https://github.com/web3well/bls-wallet/pull/597.

**Pull request merged**